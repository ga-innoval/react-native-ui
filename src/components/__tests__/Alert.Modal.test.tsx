import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { Text } from 'react-native';
import { AlertModal } from '../Alert.Modal';

describe('<AlertModal visible={true}/>', () => {
  let component: any = null;

  beforeEach(() => {
    component = render(
      <AlertModal title="Alerta" visible={true}>
        <Text>child component</Text>
      </AlertModal>
    );
  });
  it('should render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<AlertModal visible={false}/>', () => {
  it('should not render', () => {
    render(
      <AlertModal title="Titulo de alerta" visible={false}>
        <Text>Esta es una alerta que puede contener cualquier componente!</Text>
      </AlertModal>
    );
    expect(screen.queryByText(/titulo/i)).toBeNull();
  });
});
