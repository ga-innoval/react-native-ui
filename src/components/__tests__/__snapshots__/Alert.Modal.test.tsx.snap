// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertModal visible={true}/> should render correctly 1`] = `
[
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "borderRadius": 10,
            "margin": 10,
            "paddingHorizontal": 15,
            "paddingVertical": 37.5,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 15,
                "marginBottom": 25,
                "textAlign": "center",
              },
              {
                "color": "#232222",
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Alerta
        </Text>
        <Text>
          child component
        </Text>
      </View>
    </View>
  </Modal>,
  <View
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.75)",
        "bottom": 0,
        "elevation": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 1,
      }
    }
  />,
]
`;
